scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -62,13 +62,14 @@ class FormatWriter(formatOps: FormatOps) {
     sb.toString()
   }
 
-  val trailingSpace = Pattern.compile(" +$", Pattern.MULTILINE)
+  private val trailingSpace =
+    Pattern.compile("[\\s&&[^\\n]]+$", Pattern.MULTILINE)
```

Comment:
Just propose to extract it in dedicated `val horizontalWhitespaceRegex`. And what about `\r`? Windows uses it for newlines AFAIK

Also what about `[^\S\n\r]+$`? it looks a little easier. https://regex101.com/r/dktmbR/1

Reply:
will do.